provider/marathon/marathon.go

```diff
@@ -274,8 +357,24 @@ func (p *Provider) getLabel(application marathon.Application, label string) (str
 	return "", false
 }
 
-func (p *Provider) getPort(task marathon.Task, application marathon.Application) string {
-	port, err := processPorts(application, task)
+// Extract port from labels for a given service and a given marathon application
+func (p *Provider) getPort(task marathon.Task, application marathon.Application, serviceName string) string {
+	if value, ok := getServiceProperty(application, serviceName, "port"); ok {
```

Comment:
are you envisioning the signature for `processPorts` be something like: `func processPorts(application marathon.Application, task marathon.Task, portValue string, portIndex int) (int, error)`, and internally have it first see if `portValue` is a valid integer, in which case return the converted integer, otherwise attempt to extract a port at a `portIndex`?  The `""` empty `portValue` and '-1' for `portIndex` would be indicators that those don't exist...?